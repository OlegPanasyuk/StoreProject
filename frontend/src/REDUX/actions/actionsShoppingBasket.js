import { 
    ADD_GOODS_TO_BASKET, 
    SHOW_GOODS, 
    DEL_GOODS_FROM_BASKET,
    INIT_BASKET
} from './actionsTypes';

export const addGoodsToBasket = (id) => {
    return {
        type: ADD_GOODS_TO_BASKET,
        payload: {
            id
        }      
    };
};

export const showGoodsInBasketSuccess = (data) => {
    return {
        type: SHOW_GOODS.SUCCESS,
        payload: {
            data
        }
    };
};

export const deleteGoodsFromBasket = (id) => {
    return {
        type: DEL_GOODS_FROM_BASKET,
        payload: {
            id
        }
    };
};

export const initBasketFromLocalStorage = (goodsInBasket) => {
    return {
        type: INIT_BASKET,
        payload: {
            goodsInBasket
        }
    };
};