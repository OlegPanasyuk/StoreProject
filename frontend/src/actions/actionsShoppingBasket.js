import {
    ADD_GOODS_TO_BASKET,
    SHOW_GOODS,
    DEL_GOODS_FROM_BASKET,
    INIT_BASKET
} from './actionsTypes';

export const addGoodsToBasket = id => ({
    type: ADD_GOODS_TO_BASKET,
    payload: {
        id
    }
});

export const showGoodsInBasketSuccess = data => ({
    type: SHOW_GOODS.SUCCESS,
    payload: {
        data
    }
});

export const deleteGoodsFromBasket = id => ({
    type: DEL_GOODS_FROM_BASKET,
    payload: {
        id
    }
});

export const initBasketFromLocalStorage = goodsInBasket => ({
    type: INIT_BASKET,
    payload: {
        goodsInBasket
    }
});
