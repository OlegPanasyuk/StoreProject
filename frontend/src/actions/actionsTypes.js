export const ADD_GOODS_TO_BASKET = 'ADD_GOODS_TO_BASKET';

export const DEL_GOODS_FROM_BASKET = 'DEL_GOODS_FROM_BASKET';

export const INIT_BASKET = 'INIT_BASKET';

export const SHOW_GOODS = {
    SUCCESS: 'SHOW_GOODS_IN_BASKET_SUCCESS',
    FAILED: 'SHOW_GOODS_IN_BASKET_FAILED'
};

export const SAVE_BASKET = {
    SUCCESS: 'SAVE_BASKET_SUCCESS',
    FAILED: 'SAVE_BASKET_FAILED'
};

export const SHOW_HISTORY_BASKET = {
    SUCCESS: 'SHOW_HISTORY_BASKET_SUCCESS',
    FAILED: 'SHOW_HISTORY_BASKET_FAILED'
};

export const SHOW_MAIN_CONTENT = {
    SHOPPING_BASKET: 'SHOW_MAIN_CONTENT_SHOPPING_BASKET',
    USER_HISTORY: 'SHOW_MAIN_CONTENT_USER_HISTORY',
    CATALOGUE: 'SHOW_MAIN_CONTENT_CATALOGUE',
    USER_PROFILE: 'SHOW_MAIN_CONTENT_USER_PROFILE'
};

export const USER = {
    SHOW_INFO: {
        SUCCESS: 'USER_SHOW_INFO_SUCCESS',
        FAILED: 'USER_SHOW_INFO_FAILED'
    },
    SET_INFO: {
        SUCCESS: 'USER_SET_INFO_SUCCESS',
        FAILED: 'USER_SET_INFO_FAILED'
    },
    LOGIN: {
        ASK: 'USER_LOGIN_ASK'
    }
};

export const ERROR = {
    ADD: 'ERROR_ADD',
    DELETE: 'ERROR_DELETE'
};
