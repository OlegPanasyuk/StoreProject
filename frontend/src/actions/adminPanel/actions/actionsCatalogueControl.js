import { ADMIN_PANEL_CATALOGUE_CONTROL_PANEL } from './actionTypes';

export const editCatalogueItem = payload => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.EDIT,
    payload
});

export const editCatalogueItemOpen = () => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.OPEN_EDIT,
    payload: {
        show: true
    }
});

export const editCatalogueItemClose = () => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.CLOSE_EDIT,
    payload: {
        show: false
    }
});


export const addCatalogueItem = payload => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.ADD,
    payload
});

export const addCatalogueItemOpen = () => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.OPEN_ADD,
    payload: {
        show: true
    }
});

export const addCatalogueItemClose = () => ({
    type: ADMIN_PANEL_CATALOGUE_CONTROL_PANEL.ITEM.CLOSE_ADD,
    payload: {
        show: false
    }
});
