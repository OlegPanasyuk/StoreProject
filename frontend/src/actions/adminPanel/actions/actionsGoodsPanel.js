import { ADMIN_PANEL_GOODS_PANEL } from './actionTypes';

export const goodsGetSuccess = payload => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.GET.SUCCESS,
    payload
});

export const goodsGetFailed = payload => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.GET.FAILED,
    payload
});

export const goodsFilter = payload => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.FILTER.SUCCESS,
    payload
});

export const openEditGoodsItem = obj => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.EDIT.OPEN,
    payload: {
        ...obj
    }
});

export const editGoodsItem = obj => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.EDIT.SUCCESS,
    payload: {
        ...obj
    }
});

export const closeEditGoodsItem = () => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.EDIT.CLOSE
});

export const permissionToDelete = id => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.DELETE.PERMISSION,
    payload: {
        id
    }
});

export const permissionToDeleteClose = () => ({
    type: ADMIN_PANEL_GOODS_PANEL.GOODS.DELETE.CLOSE
});