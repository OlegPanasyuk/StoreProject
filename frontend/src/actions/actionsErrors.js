import { ERROR } from './actionsTypes';

export const addErrorToState = obj => ({
    type: ERROR.ADD,
    payload: {
        ...obj
    }
});

export const deleteErrorFromState = id => ({
    type: ERROR.DELETE,
    payload: {
        id
    }
});
