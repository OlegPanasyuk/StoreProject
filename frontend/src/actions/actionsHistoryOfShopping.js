import { SHOW_HISTORY_BASKET } from './actionsTypes';

export function showHistoryOfShoppping(arr) {
    return ({
        type: SHOW_HISTORY_BASKET.SUCCESS,
        payload: {
            arr
        }
    });
}

export default showHistoryOfShoppping;
