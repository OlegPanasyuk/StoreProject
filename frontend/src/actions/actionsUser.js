import { USER } from './actionsTypes';

export const setUserInfo = user => ({
    type: USER.SET_INFO.SUCCESS,
    payload: {
        user
    }
});

export const askLogin = () => ({
    type: USER.LOGIN.ASK,
    payload: {
        user: {
            role: 'Login'
        }
    }
});

export const askReg = () => ({
    type: USER.LOGIN.ASK,
    payload: {
        user: {
            role: 'Reg'
        }
    }
});
