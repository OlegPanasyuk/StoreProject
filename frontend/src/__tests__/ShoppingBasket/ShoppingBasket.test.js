import React from 'react';
import ShoppingBasketConnected, { ShoppingBasket } from '../../ShoppingBasket/ShoppingBasket';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
//import rest from 'rest';

import { addGoodsToBasket, deleteGoodsFromBasket } from '../../REDUX/actions/actionsShoppingBasket';

configure({ adapter: new Adapter() });

// import pathPrefix from 'rest/interceptor/pathPrefix';
// import errorCode from 'rest/interceptor/errorCode';
// import mime from 'rest/interceptor/mime';


// const client = rest.wrap(mime, { mime: 'application/json' })
//     .wrap(errorCode, { code: 500 })
//     .wrap(pathPrefix, { prefix: 'http://localhost:3300' });


function setupComponent(props = {
    goods: new Set([1, 2, 3]),
    goodsData: []
}) {
    let component = shallow(<ShoppingBasket {...props} />);
    return { component, props };
}

describe('ShoppingBasket --- Snapshot', () => {
    it('+++capturing Snapshot of ShoppingBasket', () => {
        const obj = setupComponent();
        expect(ShoppingBasket.propTypes.goods).toBe(PropTypes.object);
        expect(ShoppingBasket.propTypes.goodsData).toBe(PropTypes.array);
        expect(shallowToJson(obj.component)).toMatchSnapshot();
    });
});


describe('ShoppingBasket', () => {

    const initialState = {
        shoppingBasketReducers: {
            goodsInBasket: new Set([1, 2, 3]),
            goodsInBasketData: [
                {
                    idgoods: 1,
                    name: 'Honda civic',
                    description: 'Lorem ipsum dolor sit amet consectetur \
                    adipisicing elit. Blanditiis provident quo quaerat \
                    repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? Nisi, maxime a.',
                    catalogue_id_catalogue: 4,
                    price: 2000
                },
                {
                    idgoods: 7,
                    name: 'Opel Omega',
                    description: 'Lorem ipsum dolor sit amet consectetur \
                    adipisicing elit. Blanditiis provident quo quaerat \
                    repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? Nisi, maxime a.',
                    catalogue_id_catalogue: 4,
                    price: 31225
                },
                {
                    idgoods: 11,
                    name: 'Passat B7',
                    description: 'Lorem ipsum dolor sit amet consectetur \
                    adipisicing elit. Blanditiis provident quo quaerat \
                    repellat fuga debitis nemo voluptates quis voluptate, dolorem modi non? Nisi, maxime a.',
                    catalogue_id_catalogue: 4,
                    price: 121358
                }
            ]
        },
        userHeaderReducers: {
            userInfo: {
                
            }
        }
    };
    const mockStore = configureStore();
    let store, wrapper;
    beforeEach(() => {
        store = mockStore(initialState);
        wrapper = mount(<Provider store={store}><ShoppingBasketConnected /></Provider>);
    });

    it('+++ render the connected(SMART) component', () => {
        expect(wrapper.find(ShoppingBasketConnected).length).toEqual(1);
    });

    it('+++ actions', () => {
        let actions;
        store.dispatch(addGoodsToBasket(4));
        store.dispatch(deleteGoodsFromBasket(4));
        actions = store.getActions();
        expect(actions[0]).toEqual({
            type: 'ADD_GOODS_TO_BASKET',
            payload: {
                id: 4
            }
        });
        expect(actions[1]).toEqual({
            type: 'DEL_GOODS_FROM_BASKET',
            payload: {
                id: 4
            }
        });
    });

});
