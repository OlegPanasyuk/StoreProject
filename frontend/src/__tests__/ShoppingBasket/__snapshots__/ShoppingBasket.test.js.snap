// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingBascket --- Snapshot +++capturing Snapshot of ShoppingBascket 1`] = `
<Container
  as="div"
  fluid={false}
>
  <Bootstrap(Row)>
    <h3>
      Your Basket:
    </h3>
  </Bootstrap(Row)>
  <Bootstrap(Row)>
    0
  </Bootstrap(Row)>
  <Bootstrap(Row)>
    <Col
      as="div"
      className="col-10"
    >
      Total price:
    </Col>
    <Col
      as="div"
      className="shopping-basket__total-price"
    >
      0 $
    </Col>
  </Bootstrap(Row)>
  <Bootstrap(Row)>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="secondary"
    >
      Buy
    </Button>
    <Overlay
      placement="right"
      rootClose={false}
      show={false}
      target={null}
      transition={[Function]}
    >
      <Component />
    </Overlay>
  </Bootstrap(Row)>
</Container>
`;
